### DISCUSSION TITLE
What is the best way to implement Google Tag Menager?

### DISCUSSION THREAD

**Discussion by jakubhome** [vercel/next.js#72849] [Help]
### Summary

What is the most optimal way to implement Google Tag Menager in Nextjs
so that it puts the least load on the main thread and reduces Total Blocking Time?

partytown, next/third-parties?

**Comment by icyJoseph** [comment]
Sadly enough, the best thing that could happen to you, is that your GTM doesn't invoke even more annoying code... 

Use `next/third-parties`, but in general, even partytown can defeat its own purpose. Many scripts injected through GTM are UI intensive, so the overhead of wiring partytown to them, is worse. 

- https://partytown.builder.io/trade-offs#ui-intensive-third-party-scripts
